// spotify-api-client/dist/helpers.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _apiUtils = require('./apiUtils');

var _apiUtils2 = _interopRequireDefault(_apiUtils);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BASE_URL = 'https://api.spotify.com/v1';

exports.default = {
  getSearchUrl: function getSearchUrl(query, options) {
    options = options ? options : {};
    var opts = {
      'type': 'artist',
      'limit': 1,
      'offset': 0
    };
    Object.keys(options).forEach(function (key) {
      opts[key] = options[key];
    });
    return BASE_URL + '/search?q=' + encodeURI(query) + '&type=' + encodeURI(opts.type) + '&limit=' + encodeURI(opts.limit) + '&offset=' + encodeURI(opts.offset);
  },

  getTracksUrl: function getTracksUrl(artistId, country) {
    country = country ? country : 'US';
    return BASE_URL + '/artists/' + artistId + '/top-tracks?country=' + encodeURI(country);
  },

  getRequest: function getRequest(url) {
    return fetch(url, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    }).then(_apiUtils2.default.checkStatus).then(function (response) {
      return response.json();
    });
  },

  mergeTypeAndOptions: function mergeTypeAndOptions(type, options) {
    options = options ? options : {};
    return Object.assign(options, { type: type });
  }
};
