// spotify-api-client/dist/index.js
'use strict';

var _apiUtils = require('./apiUtils');

var _apiUtils2 = _interopRequireDefault(_apiUtils);

var _helpers = require('./helpers');

var _helpers2 = _interopRequireDefault(_helpers);

require('isomorphic-fetch');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

module.exports = {
  findArtist: function findArtist(query, options) {
    var opts = _helpers2.default.mergeTypeAndOptions('artist', options);
    var searchUrl = _helpers2.default.getSearchUrl(query, opts);
    return _helpers2.default.getRequest(searchUrl);
  },

  findAlbum: function findAlbum(query, options) {
    var opts = _helpers2.default.mergeTypeAndOptions('album', options);
    var searchUrl = _helpers2.default.getSearchUrl(query, opts);
    return _helpers2.default.getRequest(searchUrl);
  },

  findTrack: function findTrack(query, options) {
    var opts = _helpers2.default.mergeTypeAndOptions('track', options);
    var searchUrl = _helpers2.default.getSearchUrl(query, opts);
    return _helpers2.default.getRequest(searchUrl);
  },

  getTracks: function getTracks(artistId, country) {
    var searchUrl = _helpers2.default.getTracksUrl(artistId, country);
    return fetch(searchUrl, {
      method: 'GET',
      headers: {
        'Content-Type': 'application/json'
      }
    }).then(_apiUtils2.default.checkStatus).then(function (response) {
      return response.json();
    });
  }
};
